client/main/notification.js

```diff
@@ -0,0 +1,94 @@
+        this.bottom = (10*(this.model.attributes.index+1)+50*this.model.attributes.index);
+        this.render();
+    },
+    render: function() {
+
+        this.$el.addClass("silky-notification");
+        
+        if (this.model.attributes.visible){
+            this.$el.css("visibility", "visible");        
+        }else{
+            this.$el.css("visibility", "hidden");
+        }
+        
+        this.secondtimer = null;
+        if (this.model.attributes.timeToFade > 0){
+            setTimeout(() => {
+                this.$el.find(".silky-notification-link").addClass("silky-notification-link-fadeout");
+                this.secondtimer = setTimeout(() => {
+                    this.model.set("visible" , false);
+                    this.model.set("timeToFade" , 0);
+                    this.model.parent.remove(this);
+                }, 2000);
+            }, this.model.attributes.timeToFade);
+        }
+        
+        var html = '';
+        
+        html += '<div class="silky-notification-box">';
+        html += '    <span class="silky-notification-content">';
+        html += '       <h1>';
+        html += this.model.title;
+        html += '       </h1>';
+        html +=  (this.model.attributes.content === null) ? '':this.model.attributes.content +'</br>';
+        html +=  (this.model.attributes.description === null) ? '':this.model.attributes.description +'</br>';
+        html +=  (this.model.attributes.errorMessage === null) ? '':this.model.attributes.errorMessage +'</br>';
+        html +=  (this.model.attributes.errorCause === null) ? '':this.model.attributes.errorCause;
+        html += '    </span>';
+        if (this.model.attributes.linkText !== ''){
+            html += '    <span class="silky-notification-link">';
+            html += this.model.attributes.linkText;
+            html += '    </span>';
+        }
+        html += '</div>';
+
+        this.$el.html(html);
+        
+        
+        
+        
+        var newbottom = (10*(this.model.attributes.index+1)+50*this.model.attributes.index);
+        var diff = Math.abs(this.bottom-newbottom);
+        var $box = this.$el.find(".silky-notification-box");
+        
+        if (this.bottom === newbottom){
+            $box.css("bottom", ""+newbottom+"px");
+        }else if (this.bottom > newbottom){
+            $box.css("bottom", ""+this.bottom+"px");
+            $box.animate({bottom: "-="+diff}, "slow");
+            this.bottom = newbottom;
+        }else if (this.bottom < newbottom){
+            $box.css("bottom", ""+this.bottom+"px");
+            $box.animate({bottom: "+="+diff}, "slow");
+            this.bottom = newbottom;
+        }
+    },
+    _linkClicked : function(event){
+        this.model.linkAction();
+        if (this.model.attributes.hideOnAction){
+            this.model.set("visible" , false);
+            this.$el.css("visibility", "hidden");
+            this.render();
+            this.model.parent.remove(this);
```

Comment:
so this is OK, but you could use an observer pattern here. in software we're always trying to reduce interdependency between code. there's two object types - `Notification` and `Notifications`. Ns uses N, which means we can't use Ns without N (which is fine), but here you have N using Ns. this means that there's two way dependency. Ns can't be used without N, and N can't be used without Ns.

what you could do, is use an observer pattern, where N has some event, and Ns subscribes to that event. that way N doesn't need to have any knowledge of Ns. this also means that N can be reused elsewhere, with a different hypothetical Notifications system.

what you've done is fine, and probably you already know all this, but i did just want to point it out.

makes sense?